test/models/pet_test.rb

```diff
@@ -12,11 +12,13 @@ class PetTest < ActiveSupport::TestCase
   context "validations" do
     should validate_presence_of(:name)
     should validate_presence_of(:birth_date)
-    should validate_presence_of(:size)
     should validate_presence_of(:breed)
     should validate_presence_of(:sex)
     should validate_presence_of(:description)
     should validate_length_of(:description).is_at_most(1000)
+    should validate_presence_of(:weight_from)
+    should validate_presence_of(:weight_to)
+    should validate_presence_of(:weight_unit)
```

Comment:
Once you've updated those tests in my comment above you can add them here.

Here is the documentation of the library https://github.com/thoughtbot/shoulda-matchers with associated matchers.